.neuro/live-s3.yml

```
@@ -0,0 +1,314 @@
+    mount: /usr/project/src
+    local: src
+  config:
+    remote: storage:/$[[ project.owner ]]/$[[ flow.flow_id ]]/config
+    mount: /usr/project/config
+    local: config
+  notebooks:
+    remote: storage:/$[[ project.owner ]]/$[[ flow.flow_id ]]/notebooks
+    mount: /usr/project/notebooks
+    local: notebooks
+
+images:
+  train:
+    ref: image:/$[[ project.owner ]]/$[[ flow.flow_id ]]/train:21.5.13
+    dockerfile: $[[ flow.workspace ]]/Dockerfile
+    context: $[[ flow.workspace ]]/
+    build_preset: cpu-medium
+  seldon:
+    ref: image:/$[[ project.owner ]]/$[[ flow.flow_id ]]/seldon:20.12.16
+    dockerfile: $[[ flow.workspace ]]/seldon/seldon.Dockerfile
+    context: $[[ flow.workspace ]]/
+    build_preset: cpu-medium
+  label_studio:
+    ref: image:/$[[ project.owner ]]/$[[ flow.flow_id ]]/label_studio:21.1.18
+    dockerfile: $[[ flow.workspace ]]/label_studio/label_studio.Dockerfile
+    context: $[[ flow.workspace ]]/label_studio
+    build_preset: cpu-medium
+
+jobs:
+  prepare_remote_dataset:
+    name: $[[ flow.title ]]-prepare-dataset
+    image: neuromation/neuro-extras:20.12.15
+    detach: False
+    volumes:
+      - ${{ volumes.remote_dataset.ref_rw }}
+    params:
+      dataset_path:
+        default: http://vision.stanford.edu/aditya86/ImageNetDogs/images.tar
+        descr: URL of dataset to download
+    bash: |
+      DST=$(basename ${{ params.dataset_path }} .tar)
+      if [ ! -e ${{ volumes.remote_dataset.mount }}/$DST ]; then
+        neuro-extras data cp --extract --use-temp-dir ${{ params.dataset_path }} ${{ volumes.remote_dataset.mount }}/$DST;
+      else
+        echo "Dataset is already downloaded, skipping."
+      fi
+
+  extend_data:
+    name: $[[ flow.title ]]-extend-data
+    image: $[[ images.train.ref ]]
+    detach: False
+    pass_config: True
+    volumes:
+      - ${{ volumes.remote_dataset.ref_ro }}
+      - secret:gh-rsa:/root/.ssh/id-rsa
+    env:
+      PROJECT: /usr/project
+      PYTHONPATH: /usr/project
+    params:
+      extend_dataset_by:
+        default: "1"
+        descr: |
+          How many new images to add into current dataset.
+    bash: |
+      git clone -b ${PROJECT_GIT_BRANCH} ${PROJECT_GIT_REPO} ${PROJECT} > /dev/null 2>&1
+
+      export DATA_PATH=${PROJECT}/data/images
+
+      # Init S3
+      mkdir -p $DATA_PATH
+      [[ -z $(neuro blob ls blob:${BUCKET_NAME}) ]] && echo "Bucket is empty" \
+        || neuro blob cp -rT blob:${BUCKET_NAME} $(dirname ${DATA_PATH}) # download data if the bucket is not empty
+
+      echo "Extending dataset"
+      python ${PROJECT}/label_studio/extend_dataset.py \
+        --cur_dataset ${DATA_PATH} \
+        --full_dataset ${{ volumes.remote_dataset.mount }}/images/Images/ \
+        --nmber_of_imgs ${{ params.extend_dataset_by }}
+
+      # Upload dataset to S3"
+      neuro blob cp -ru ${DATA_PATH} blob:${BUCKET_NAME}
+
+      # Validate dataset:
+      neuro blob ls -r blob:${BUCKET_NAME}
```

```suggestion
      neuro blob ls -r blob:${BUCKET_NAME} | tee
```
funny bug: if we have more items than window height, the pagination will be enabled and the job will appear to be frozen